tests/dataplane-performance/manifests/cafe-routes.yaml

```diff
@@ -0,0 +1,60 @@
+apiVersion: gateway.networking.k8s.io/v1beta1
+kind: HTTPRoute
+metadata:
+  name: coffee
+spec:
+  parentRefs:
+  - name: cafe
+  hostnames:
+  - cafe.example.com
+  rules:
+  - matches:
+    - path:
```

Comment:
For the path-only rule, can we use a different path than coffee?

Otherwise, every route tests the njs code:
```
server {
    listen 80;

    server_name cafe.example.com;


    location /coffee_prefix_route0 {
        internal;

        proxy_set_header Host $gw_api_compliant_host;
        proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
        proxy_http_version 1.1;
        proxy_set_header Upgrade $http_upgrade;
        proxy_set_header Connection $connection_upgrade;
        proxy_pass http://invalid-backend-ref$request_uri;
    }

    location /coffee_prefix_route1 {
        internal;

        proxy_set_header Host $gw_api_compliant_host;
        proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
        proxy_http_version 1.1;
        proxy_set_header Upgrade $http_upgrade;
        proxy_set_header Connection $connection_upgrade;
        proxy_pass http://invalid-backend-ref$request_uri;
    }

    location /coffee_prefix_route2 {
        internal;

        proxy_set_header Host $gw_api_compliant_host;
        proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
        proxy_http_version 1.1;
        proxy_set_header Upgrade $http_upgrade;
        proxy_set_header Connection $connection_upgrade;
        proxy_pass http://invalid-backend-ref$request_uri;
    }

    location /coffee/ {
        set $http_matches "[{\"redirectPath\":\"/coffee_prefix_route0\",\"headers\":[\"version:v2\"]},{\"redirectPath\":\"/coffee_prefix_route1\",\"params\":[\"TEST=v2\"]},{\"redirectPath\":\"/coffee_prefix_route2\",\"any\":true}]";
        js_content httpmatches.redirect;

    }

    location = /coffee {
        set $http_matches "[{\"redirectPath\":\"/coffee_prefix_route0\",\"headers\":[\"version:v2\"]},{\"redirectPath\":\"/coffee_prefix_route1\",\"params\":[\"TEST=v2\"]},{\"redirectPath\":\"/coffee_prefix_route2\",\"any\":true}]";
        js_content httpmatches.redirect;

    }

    location /tea_prefix_route0 {
        internal;

        proxy_set_header Host $gw_api_compliant_host;
        proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
        proxy_http_version 1.1;
        proxy_set_header Upgrade $http_upgrade;
        proxy_set_header Connection $connection_upgrade;
        proxy_pass http://invalid-backend-ref$request_uri;
    }

    location /tea_prefix_route1 {
        internal;

        proxy_set_header Host $gw_api_compliant_host;
        proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
        proxy_http_version 1.1;
        proxy_set_header Upgrade $http_upgrade;
        proxy_set_header Connection $connection_upgrade;
        proxy_pass http://invalid-backend-ref$request_uri;
    }

    location /tea/ {
        set $http_matches "[{\"method\":\"POST\",\"redirectPath\":\"/tea_prefix_route0\"},{\"method\":\"GET\",\"redirectPath\":\"/tea_prefix_route1\"}]";
        js_content httpmatches.redirect;

    }

    location = /tea {
        set $http_matches "[{\"method\":\"POST\",\"redirectPath\":\"/tea_prefix_route0\"},{\"method\":\"GET\",\"redirectPath\":\"/tea_prefix_route1\"}]";
        js_content httpmatches.redirect;

    }

    location / {
        return 404 "";

    }
```

and it would be good to be able to compare the njs routes to the native nginx.conf routes.

Reply:
Good catch, thanks!!